aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Orchestrator.java

```diff
@@ -62,7 +60,7 @@
     private final MutationProcessor mutationProcessor;
     private final QueryPredicateProvider queryPredicateProvider;
     private final StorageObserver storageObserver;
-    private final Supplier<Mode> targetMode;
+    private final boolean apiSyncEnabled;
```

Comment:
I was initially a bit confused by `targetMode`.   I first had thought it was the end state for any transition and would only differ from `currentMode` if a transition was in progress, but that is not the case.  `targetMode` never changes.  For example, during a transition from `SYNC_VIA_API` to `LOCAL_ONLY` (due to loss of internet connection), the targetMode is still `SYNC_VIA_API`.  It is more of a _penultimate_ target mode - the desired mode in the absence of errors.  

You do have a good counterpoint though that using an enum instead of a boolean is a little better semantically. 
 `apiSyncEnabled` doesn't seem ambiguous to me, but maybe that's just because I wrote it :).    

Thinking further though - I'm about to add in logic to retry after the orchestrator stops,  However, we only want to retry if it stopped due to an internal error, not if the customer explicitly calls `stop()`.   I'm going to need some state to distinguish between these cases.

Here's what I'm thinking now:
 - Switch back to using the `targetMode` enum, instead the `apiSyncEnabled` boolean I added.
 - When customer calls start(), set targetMode to `LOCAL_ONLY` or `SYNC_VIA_API`.  This would be passed in as a parameter to `Orchestrator.start()`.  
 - `targetMode` will no longer be set on `Orchestrator` initialization.
 - When customer calls stop(), set targetMode to `STOPPED`

Once retry logic is added, the purpose of `targetMode` will be more clear.